### PR TITLE
OOP school library: add basic UI

### PR DESCRIPTION
**Pull request by luccamicale**
## School Library Basic UI 💯 

##https://github.com/luccamicale/OOP-Library-School 👍 

## Checklist before requesting a review
- [x] I have performed a self-review of my code
- [x] Created app.rb and main.rb files
- [x] Created methods to:
      List all books.
      List all people.
      Create a person (teacher or student, not a plain Person).
      Create a book.
      Create a rental.
      List all rentals for a given person id.
- [x] Added an interface class 🔢 🥇

### CODE REVIEW

**Review by victorokoroji — CHANGES_REQUESTED** [review]
Hi @luccamicale, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## ❌ Invalid Code Review Request

You have submitted a project, that does not contain the `future-feature` section in your readme. Kindly add at least 2 future features you will potentially add to your project in the `future-feature` section of your readme 👍🏽 

- I will also recommend you also always capitalize the first letter of your commit messages and it should be written in the imperative form. e.g `Add book.rb file`

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Comment by luccamicale** [comment]
Hi @vickymarz I modified the readme. Now I added this part that you ask me.

File: app.rb [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+    @people = []
+    @rentals = []
+  end
+
+  def run
+    prompt_user
+  end
+
+  # List all books
+  def list_all_books
```
Comment (jasonfelice) [inline app.rb:21]: - [ ] Kindly fix the following error on List all Books

image

File: app.rb [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+    else
+      puts @books.count <= 1 ? "\n#{@books.count} Book\n" : "\n#{@books.count} Book's \n"
+      @books.each_with_index do |book, index|
+        puts "#{index + 1} Title: '#{book.title}', Author: #{book.author}"
+      end
+    end
+  end
+
+  # List Person: Student/Teacher
+  def list_all_people
```
Comment (jasonfelice) [inline app.rb:33]: - [ ] Same error on List all People. Please fix this error. 😄

image

**Review by jasonfelice — CHANGES_REQUESTED** [review]
Hi @luccamicale, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

## Highlights
No Linter Errors✔️

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by luccamicale** [comment]
Hi @jasonfelice I dont understand how to solve that. Can you help me please?

**Comment by luccamicale** [comment]
I fix the two errors in my code. the method is the same in the file main and app. Now my project should be approved!!! 💯 👍 🔢

File: app.rb (hunk 2 above)
Comment (SajjadAhmad14) [inline app.rb:33]: Good job on making the progress! But there is still an error when user tries to list all people. Kindly see the error and fix it.

people

**Review by SajjadAhmad14 — CHANGES_REQUESTED** [review]
## Required Changes ♻️
> "Change is inevitable. Change is constant. -Benjamin Disraeli"
## Hi Lucca,
### Great work on making the changes requested by a previous reviewer 👏 
### You've done well implementing some of the requested changes, but there are still some which aren't been addressed yet.
## To Highlight 🎉
No linter errors. :heavy_check_mark:
Git flow is followed. :heavy_check_mark:
## To Improve 🙏

- [x] Please check inline comments under the review

## Optional suggestions
_Comments with the [OPTIONAL] prefix are not a must to implement however, It's strongly recommended to work on them as they can make your code better._

## Happy coding!👏👏👏

#### Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
#### Please remember to tag me in your question so I can receive the notification.
**_Note: Please do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Comment by luccamicale** [comment]
I fix the only problem. the last reviewer help me with the problem and told me that removing the person.type the problem it is solve. Now my project should be approved!!!! 💯 🥇

**Review by mwanawabangona — APPROVED** [review]
Hi @luccamicale,

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:
Congratulations!  


## Highlights
*   No linter errors :heavy_check_mark:
*   Repo looks professional :briefcase:
*   Solid pull request scroll
*   Good quality commits
*   Good  job implementing the requested chnges!! :clap: :clap: :clap:

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**